packages/core/src/common/entity-preferences/store.ts

```diff
@@ -20,30 +20,36 @@ export interface EntityPreferencesStoreModel {
   entities?: [string, EntityPreferencesModel][];
 }
 
-export class EntityPreferencesStore extends BaseStore<EntityPreferencesStoreModel> {
-  @observable readonly preferences = observable.map<string, PartialDeep<EntityPreferencesModel>>();
+export interface EntityPreferencesStoreDependencies {
+  createBaseStore: CreateBaseStore;
+  readonly storeMigrationVersion: string;
+}
+
+export class EntityPreferencesStore {
+  private readonly store: BaseStore<EntityPreferencesStoreModel>;
 
-  constructor(deps: BaseStoreDependencies) {
-    super(deps, {
+  readonly preferences = observable.map<string, PartialDeep<EntityPreferencesModel>>();
+
+  constructor(private readonly dependencies: EntityPreferencesStoreDependencies) {
+    this.store = this.dependencies.createBaseStore({
       configName: "lens-entity-preferences-store",
+      projectVersion: this.dependencies.storeMigrationVersion,
+      fromStore: action(({ entities = [] }) => {
+        this.preferences.replace(entities);
+      }),
+      toJSON: () => ({
+        entities: this.preferences.toJSON(),
+      }),
     });
-
-    makeObservable(this);
   }
 
-  @action
   mergePreferences(entityId: string, preferences: PartialDeep<EntityPreferencesModel>): void {
-    this.preferences.set(entityId, merge(this.preferences.get(entityId), preferences));
-  }
-
-  @action
-  protected fromStore(data: EntityPreferencesStoreModel): void {
-    this.preferences.replace(data.entities ?? []);
+    runInAction(() => {
+      this.preferences.set(entityId, merge(this.preferences.get(entityId), preferences));
+    });
```

Comment:
Why do we need to "run in action" that single observable's update? or updating any map/set or single in general observable unit. Afaik `runInAction` needs to optimize update/reactions and do it in single "transaction" when updating multiple observables within a scope, e.g. 

```ts
// that would make sense for me for example
runInAction(() => {
      this.preferences.set(entityId, merge(this.preferences.get(entityId), preferences));
      this.somethingObservableToo = "UPDATED!";
    });
```

Reply:
Yes `runInAction` is a way to specify a transaction. The reason why updating a single observable doesn't normally need to also be within an action is because we currently have `enforceActions: false` as part of our mobx configuration.